baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/ForbidJavaxParameterType.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.palantir.errorprone.ForbidJavax;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.ClassSymbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Symbol.TypeSymbol;
+import com.sun.tools.javac.code.Symbol.VarSymbol;
+import com.sun.tools.javac.code.Type;
+import java.lang.annotation.Annotation;
+import java.util.List;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.ERROR,
+        summary = "Supplying an object which uses legacy javax types, such as javax.ws.rs to a\n"
+                + "method which requires newer jakarta types is a runtime error. This check ensures\n"
+                + "that you only supply proper types to these methods which generally just take an\n"
+                + "untyped Object. There is no auto-fix for this check, you must fix it manually")
+public final class ForbidJavaxParameterType extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final Matcher<ClassTree> HAS_JAXRS_ANNOTATIONS = Matchers.hasMethod(Matchers.anyOf(
+            Matchers.hasAnnotation("javax.ws.rs.GET"),
+            Matchers.hasAnnotation("javax.ws.rs.POST"),
+            Matchers.hasAnnotation("javax.ws.rs.DELETE"),
+            Matchers.hasAnnotation("javax.ws.rs.PUT"),
+            Matchers.hasAnnotation("javax.ws.rs.HEAD"),
+            Matchers.hasAnnotation("javax.ws.rs.OPTIONS")));
```

Comment:
iirc these are all annotated with something along the lines of `javax.ws.rs.Method`, we could search for any annotation from `javax.ws.rs.*` or annotated with the jaxrs meta-annotations

Reply:
I haven't been able to find any clean method of doing such a thing....I could probably write something custom though, it'll just be a little messy.

Reply:
I've created a PR with an example that fails: #2367

Reply:
Operating directly on the result type symbol will give us what we want, that provides the set of class+runtime retained annotations, list of methods and their annotations/etc